docs/semgrep-secrets/historical-scanning.md

```diff
@@ -107,8 +107,7 @@ Historical scan findings are not automatically marked as **Fixed**. To triage a
 
 - Historical scanning can slow down scan times. Depending on the size of your repository history, scans can finish under 5 minutes to more than 60 minutes for extreme cases.
 - Within Semgrep AppSec Platform, historical scan findings are not automatically marked as **Fixed**. Findings can only exist in two states: `Open` or `Ignored`. Because Semgrep scans do not automatically detect historical findings as fixed, you must manually rotate and triage the secret as `Ignored`.
-- A finding can show up twice in the CLI with historical scanning enabled: the HEAD commit in the regular Secrets scan and another commit in the historical scan.
-    - If findings are sent to Semgrep AppSec Platform, they are deduplicated and appear as a **regular finding**, not a historical finding.
+- With historical scanning enabled, the CLI may display duplicate instances of the same secret finding if it occurs across various commits or branches. This is to ensure thorough analysis by presenting all occurrences. However, when utilizing the Semgrep AppSec Platform, findings from historical scans are deduplicated. Consequently, you'll encounter only one entry for each distinct secret, irrespective of its multiple appearances in your Git history..
```

Comment:
In the current wording, I aim to clarify that while the CLI displays duplicate instances of the same secret finding across different commits or branches, the UI only presents a single, distinct secret. This single instance typically originates from the main or default branch. So users can   clearly understand the differences

Reply:
I think a combination of your clarification and explanation is the clearest I've seen this explained; maybe something like

> With historical scanning, the CLI displays duplicate instances of the same secret finding across different commits or branches, while Semgrep AppSec Platform only presents a single, distinct secret. This single instance typically originates from the main or default branch.
>
> The duplicate instances displayed by the CLI helps ensure thorough analysis by presenting all occurrences of the secret, while Semgrep AppSec Platform deduplicates repeated findings.

Reply:
I think this looks good to me!

Reply:
Hmm, I'm a little meh on this. I don't think we do it for thoroughness, although I may be wrong (I think it's just hard on the CLI so we decided to do it on the app side). The phrasing is also a bit vague, the correct operation is that it would show at most twice, once at the commit where it's added (historical), and once in the non-historical if it's still present.

How about:

> When performing historical scanning, the CLI output displays secrets that are still present in the current version of the code twice: once at the commit where they were originally added, and once at the current commit from the standard Secrets scan.

> The Semgrep AppSec Platform deduplicates the two findings, and displays the secret as a current secret, rather than a historical secret.

Reply:
I've got some issues with the other bullet points in this section too now that I look, but I'll leave that for another PR. :)

Reply:
Hi @armchairlinguist 
Looking at this from a not so technical user perspective reading the doc, I might find this wording  abit confusing.
Let me know if this also brings clarity:

> When performing historical scanning with the CLI, you might encounter duplicate instances of the same secret finding across different commits or branches. This behavior ensures a thorough analysis by presenting all occurrences of the secret, regardless of their location in the code's history. In contrast, the Semgrep AppSec Platform takes a deduplication approach. It presents a single, distinct secret, typically originating from the main or default branch. This streamlined view keeps the findings concise and focuses on currently active secrets within your codebase.

Reply:
I'm not sure if I was clear in my previous note, but I don't think this statement is correct, because I don't think it's true that we do this to be thorough. The tone is also trying too hard to sell me on the behavior, instead of just describing it.

Reply:
Hi folks. I apologize for stepping in, I dont mean to step on @khorne3 's toes here (and will not have any stamping authority ofc)

## Please note that this is specifically for the case of dupes where the secret is present in HEAD **AND** previous commit(s)

Indeed, AFAIK we don't present the "dupes" to be thorough, it's just how the CLI works (i.e. we didn't put any dedupe logic in it); it wasn't intentional. If anything, the dupes are... almost noise, because it presents, let's say 9999 instances of "ABCD" (I am exaggerating but you get the idea) but the only action to take is to remove "ABCD" from 1 location in history (the present, HEAD), because it is not a "historical" finding, but a current one. 

As a general illustration, to give you an idea of how much noise could be generated from historical scanning in git IN GENERAL, outside of secrets, you can try running the following git command that searches for a known string in any repository. In this example it searches through the entire semgrep-docs repo for "You must have Python 3.8"

```bash
git rev-list --all | (
    while read revision; do
        git grep -F 'You must have Python 3.8' $revision
    done
)
```

We then get something like this:
![image](https://github.com/semgrep/semgrep-docs/assets/20766840/fb06b378-24ae-456a-b211-6e11fb2f643a)

Which is wayyyy too many results for it to be useful to a human.

I hope this helps and if you would like a call with me or Sal (the engineer on historical scanning) please let me know.

Reply:
[Slack response from Engineering team](https://semgrepinc.slack.com/archives/C05C6JM7K2L/p1715787192852819)

I'm not the writer assigned to review this, but I would suggest at least **removing specifically the bit that we detect these duplicates for thoroughness in the CLI (we don't)**. Just describe the behavior.

cc @khorne3

Reply:
Okay that's fine then would make the commit with [this](https://github.com/semgrep/semgrep-docs/pull/1546#discussion_r1600386777)

Reply:
Sounds good. I cleaned it up a bit:

> With historical scanning enabled, the CLI output displays secrets still present in the current version of the code twice: once at the commit where they were initially added and once at the current commit from the standard Secrets scan. Semgrep AppSec Platform deduplicates the two findings and displays the secret as a current rather than a historical one.

If that looks good, I'll commit the change.

Reply:
Yes I'm good with that, thank you!